tests/e2e/vmlistener_test.go

```diff
@@ -186,6 +192,102 @@ func (s *VMListenerTestParams) TestFailoverAcrossVM(c *C) {
 	misc.LogTestEnd(c.TestName())
 }
 
+// TestVolumeAttachedForHostdRestart - verify an in use volume
+// remains attached when hostd is restarted.
+// 1. Restart hostd
+// 2. Verify verify volume stays as attached
+// 3. Kill the vm
+// 4. Status of the volume should be detached.
+func (s *VMListenerTestParams) TestVolumeAttachedForHostdRestart(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// 1. Restart hostd
+	out, err := ssh.InvokeCommand(s.esx, restartHostd)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Give hostd time to start up
+	misc.SleepForSec(30)
```

Comment:
Can't we call "/etc/init.d/hostd status"? I'm fine if we want to address this in a follow-up change, but I do suggest to avoid blind sleeping in the test logic, for the reason that I have explained above.